optimization.py

```diff
@@ -132,20 +132,19 @@ def merge_datasets(self):
                             merged_graph.children.append(child)
         self.graphs = merged_graphs
         logger.debug('%%%%%%%%%% Merging datasets: DONE')
+        logger.debug('Merged graphs:\n{}'.format(self.get_pretty_printed_merged_graphs()))
 
     def optimize_selections(self):
         logger.debug('%%%%%%%%%% Optimizing selections:')
         for merged_graph in self.graphs:
             self._merge_children(merged_graph)
         logger.debug('%%%%%%%%%% Optimizing selections: DONE')
 
-    def print_merged_graphs(self):
+    def get_pretty_printed_merged_graphs(self):
         def call_node_rec(nd):
             return PrintedNode(nd.__repr__())([call_node_rec(child) for child in nd.children])
         fancy_trees = [call_node_rec(graph) for graph in self.graphs]
-        logger.debug('Merged graphs:')
-        logger.debug('\n\n'.join([
-            drawTree2(False)(False)(t) for t in fancy_trees]))
+        return '\n\n'.join([drawTree2(False)(False)(t) for t in fancy_trees])
```

Comment:
What are the dependencies here? Sry haven't followed the implementation here.

Reply:
no dependencies, I included the thing in the printing.py module

Reply:
Alright, thanks!